### PR TITLE
Add support for pre-commit

### PR DESCRIPTION
**Pull request by damienrj**
This adds support for https://pre-commit.com/ to Hermit.

### CODE REVIEW

File: pre-commit.hcl [hunk 1]
```diff
@@ -0,0 +1,18 @@
+description = "Pre-commit is a multi-language package manager for pre-commit hooks."
+binaries = ["pre-commit"]
+test = "pre-commit --version"
+source = "https://github.com/pre-commit/pre-commit/releases/download/v${version}/pre-commit-${version}.pyz"
+strip = 1
```
Comment (alecthomas) [inline pre-commit.hcl:5]: This is redundant now.

File: pre-commit.hcl [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+runtime-dependencies = ["python3@3.9"]
+provides=["pre-commit"]
```
Comment (alecthomas) [inline pre-commit.hcl:7]: This should only be used for "virtual" packages such as "jre", and should not match an existing "real" package. You'll have to remove it.

**Review by alecthomas — APPROVED** [review]

File: pre-commit.hcl (hunk 2 above)
Comment (alecthomas) [reply pre-commit.hcl:7]: Want to fix these up and merge?

File: pre-commit.hcl (hunk 1 above)
Comment (damienrj) [reply pre-commit.hcl:5]: ✅

File: pre-commit.hcl (hunk 2 above)
Comment (damienrj) [reply pre-commit.hcl:7]: ✅

**Pull request merged**